catalog/resource_sql_table.go

```diff
@@ -294,8 +308,24 @@ func (ti *SqlTableInfo) deleteTable() error {
 
 func (ti *SqlTableInfo) applySql(sqlQuery string) error {
 	log.Printf("[INFO] Executing Sql: %s", sqlQuery)
-	r := ti.exec.Execute(ti.ClusterID, "sql", sqlQuery)
+	if ti.WarehouseID != "" {
+		execCtx, cancel := context.WithTimeout(context.Background(), time.Second*10)
```

Comment:
This is a very short timeout and may cause issues. For example, if there is a transient error and the request can be retried, this may be less forgiving. Even for requests which take a full 10 seconds,  We may want to use a longer timeout here, at least as long as `DatabricksClient.retryTimeout` (which defaults to 300 seconds). It's a bit screwy for the command execution API, which today captures the context in the `CommandsAPI` struct directly and doesn't set any timeouts.